core/src/main/java/org/wso2/msf4j/internal/MSF4JMessageProcessor.java

```diff
@@ -167,7 +256,275 @@ public void setTransportSender(TransportSender transportSender) {
+                                }
+                            }
+                        } else if (parameter.getType() == Session.class) {
+                            parameterList.add(session);
+                        } else {
+                            parameterList.add(null);
+                        }
+                    }
+            );
+            executeMethod(method, endpoint, parameterList, session);
+        } catch (Throwable throwable) {
+            handleError(textCarbonMessage, throwable, routableEndpoint, session);
+        }
+    }
+
+    /*
+    Handle Binary WebSocket Message
+     */
+
+    private void handleBinaryWebSocketMessage(BinaryCarbonMessage binaryCarbonMessage,
+                                              PatternPathRouter.RoutableDestination<Object>
+                                                      routableEndpoint, Session session) {
+        Object webSocketEndpoint = routableEndpoint.getDestination();
+        Map<String, String> paramValues = routableEndpoint.getGroupNameValues();
+        Method method = new EndpointDispatcher().getOnBinaryMessageMethod(webSocketEndpoint);
+        try {
+            List<Object> parameterList = new LinkedList<>();
+            Arrays.stream(method.getParameters()).forEach(
+                    parameter -> {
+                        if (parameter.getType() == ByteBuffer.class) {
+                            parameterList.add(binaryCarbonMessage.readBytes());
+                        } else if (parameter.getType() == byte[].class) {
+                            ByteBuffer buffer = binaryCarbonMessage.readBytes();
+                            byte[] bytes = new byte[buffer.capacity()];
+                            for (int i = 0; i < buffer.capacity(); i++) {
+                                bytes[i] = buffer.get();
+                            }
+                            parameterList.add(bytes);
+                        } else if (parameter.getType() == boolean.class) {
+                            parameterList.add(binaryCarbonMessage.isFinalFragment());
+                        } else if (parameter.getType() == Session.class) {
+                            parameterList.add(session);
+                        } else if (parameter.getType() == String.class) {
+                            PathParam pathParam = parameter.getAnnotation(PathParam.class);
+                            if (pathParam != null) {
+                                parameterList.add(paramValues.get(pathParam.value()));
+                            } else {
+                                throw new IllegalArgumentException("String parameters without" +
+                                                                                 " @PathParam annotation");
+                            }
+                        } else {
+                            parameterList.add(null);
```

Comment:
IMO shouldn't we add new Object() here to avoid possible null pointer exceptions. Consider using new Object()